libraries/AC_SafeRTL/AC_SafeRTL.cpp

```diff
@@ -0,0 +1,573 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "AC_SafeRTL.h"
+
+extern const AP_HAL::HAL& hal;
+
+const AP_Param::GroupInfo AC_SafeRTL::var_info[] = {
+    // @Param: ACCURACY
+    // @DisplayName: SafeRTL _accuracy
+    // @Description: SafeRTL _accuracy. The distance flown (in meters) before a new point is dropped.
+    // @Range: 0 10
+    // @User: Advanced
+    AP_GROUPINFO("ACCURACY", 0, AC_SafeRTL, _accuracy, SAFERTL_ACCURACY_DEFAULT),
+
+    // @Param: PATH_LEN
+    // @DisplayName: SafeRTL path length
+    // @Description: SafeRTL path length. The maximum number of points saved by SafeRTL. Set to 0 to disable SafeRTL.
```

Comment:
Sure, added, "100 points consumed 3k of memory" to parameter description in follow up commit.